C Compiler error

**Environment**
Ubuntu 20.04 64bit
Crystal 1.2.2 [6529d725a] (2021-11-10)

**Installed pre-requisites**
`sudo apt install pkg-config build-essential libgtk-3-dev libwebkit2gtk-4.0-dev`

---

To execute the first example on readme, on a void directory, I created both `shard.yml` and a crystal file `url.cr` with:

**shard.yml**
```yml
name: webview-test
version: 0.1.0
dependencies:
  webview:
    github: naqvis/webview
```

**url.cr**
```crystal
require "webview"

wv = Webview.window(640, 480, Webview::SizeHints::NONE, "Hello WebView", "http://crystal-lang.org")
wv.run
wv.destroy
```

And `shards install` goes well:

```sh
Resolving dependencies
Fetching https://github.com/naqvis/webview.git
Installing webview (0.1.4)
Postinstall of webview: make
Writing shard.lock
``` 

---

**The C compiler error**

But when trying to run the example with `crystal run url.cr`, I get:

**Final error**

```sh
Error: execution of command failed with code: 1: `cc "${@}" -o /home/coghi/.cache/crystal/crystal-run-url.tmp  -rdynamic -L/usr/bin/../lib/crystal `command -v pkg-config > /dev/null && pkg-config --cflags --libs gtk+-3.0 webkit2gtk-4.0` /home/coghi/Projetos/webview-tests/lib/webview/src/../ext/libwebview.a -lstdc++ -lpcre -lm -lgc -lpthread -levent -lrt -ldl`
```

**Long error:**

<details><summary>Open here:</summary><p>

```sh
/usr/bin/ld: /home/coghi/Projetos/webview-tests/lib/webview/src/../ext/libwebview.a(webview.o): in function `webview::gtk_webkit_engine::gtk_webkit_engine(bool, void*)::{lambda(_WebKitUserContentManager*, _WebKitJavascriptResult*, void*)#2}::operator()(_WebKitUserContentManager*, _WebKitJavascriptResult*, void*) const':
webview.cc:(.text._ZZN7webview17gtk_webkit_engineC4EbPvENKUlP25_WebKitUserContentManagerP23_WebKitJavascriptResultS1_E0_clES3_S5_S1_[_ZZN7webview17gtk_webkit_engineC4EbPvENKUlP25_WebKitUserContentManagerP23_WebKitJavascriptResultS1_E0_clES3_S5_S1_]+0x3c): undefined reference to `webkit_javascript_result_get_js_value'
/usr/bin/ld: webview.cc:(.text._ZZN7webview17gtk_webkit_engineC4EbPvENKUlP25_WebKitUserContentManagerP23_WebKitJavascriptResultS1_E0_clES3_S5_S1_[_ZZN7webview17gtk_webkit_engineC4EbPvENKUlP25_WebKitUserContentManagerP23_WebKitJavascriptResultS1_E0_clES3_S5_S1_]+0x4c): undefined reference to `jsc_value_to_string'
/usr/bin/ld: webview.cc:(.text._ZZN7webview17gtk_webkit_engineC4EbPvENKUlP25_WebKitUserContentManagerP23_WebKitJavascriptResultS1_E0_clES3_S5_S1_[_ZZN7webview17gtk_webkit_engineC4EbPvENKUlP25_WebKitUserContentManagerP23_WebKitJavascriptResultS1_E0_clES3_S5_S1_]+0xb1): undefined reference to `g_free'
/usr/bin/ld: /home/coghi/Projetos/webview-tests/lib/webview/src/../ext/libwebview.a(webview.o): in function `webview::gtk_webkit_engine::gtk_webkit_engine(bool, void*)':
webview.cc:(.text._ZN7webview17gtk_webkit_engineC2EbPv[_ZN7webview17gtk_webkit_engineC5EbPv]+0x4e): undefined reference to `gtk_init_check'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engineC2EbPv[_ZN7webview17gtk_webkit_engineC5EbPv]+0x71): undefined reference to `gtk_window_new'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engineC2EbPv[_ZN7webview17gtk_webkit_engineC5EbPv]+0x9d): undefined reference to `g_type_check_instance_cast'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engineC2EbPv[_ZN7webview17gtk_webkit_engineC5EbPv]+0xc2): undefined reference to `g_signal_connect_data'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engineC2EbPv[_ZN7webview17gtk_webkit_engineC5EbPv]+0xc7): undefined reference to `webkit_web_view_new'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engineC2EbPv[_ZN7webview17gtk_webkit_engineC5EbPv]+0xd4): undefined reference to `webkit_web_view_get_type'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engineC2EbPv[_ZN7webview17gtk_webkit_engineC5EbPv]+0xea): undefined reference to `g_type_check_instance_cast'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engineC2EbPv[_ZN7webview17gtk_webkit_engineC5EbPv]+0xf2): undefined reference to `webkit_web_view_get_user_content_manager'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engineC2EbPv[_ZN7webview17gtk_webkit_engineC5EbPv]+0x12e): undefined reference to `g_signal_connect_data'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engineC2EbPv[_ZN7webview17gtk_webkit_engineC5EbPv]+0x141): undefined reference to `webkit_user_content_manager_register_script_message_handler'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engineC2EbPv[_ZN7webview17gtk_webkit_engineC5EbPv]+0x194): undefined reference to `gtk_widget_get_type'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engineC2EbPv[_ZN7webview17gtk_webkit_engineC5EbPv]+0x1aa): undefined reference to `g_type_check_instance_cast'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engineC2EbPv[_ZN7webview17gtk_webkit_engineC5EbPv]+0x1b2): undefined reference to `gtk_container_get_type'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engineC2EbPv[_ZN7webview17gtk_webkit_engineC5EbPv]+0x1c8): undefined reference to `g_type_check_instance_cast'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engineC2EbPv[_ZN7webview17gtk_webkit_engineC5EbPv]+0x1d3): undefined reference to `gtk_container_add'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engineC2EbPv[_ZN7webview17gtk_webkit_engineC5EbPv]+0x1d8): undefined reference to `gtk_widget_get_type'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engineC2EbPv[_ZN7webview17gtk_webkit_engineC5EbPv]+0x1ee): undefined reference to `g_type_check_instance_cast'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engineC2EbPv[_ZN7webview17gtk_webkit_engineC5EbPv]+0x1f6): undefined reference to `gtk_widget_grab_focus'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engineC2EbPv[_ZN7webview17gtk_webkit_engineC5EbPv]+0x201): undefined reference to `webkit_web_view_get_type'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engineC2EbPv[_ZN7webview17gtk_webkit_engineC5EbPv]+0x217): undefined reference to `g_type_check_instance_cast'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engineC2EbPv[_ZN7webview17gtk_webkit_engineC5EbPv]+0x21f): undefined reference to `webkit_web_view_get_settings'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engineC2EbPv[_ZN7webview17gtk_webkit_engineC5EbPv]+0x234): undefined reference to `webkit_settings_set_enable_write_console_messages_to_stdout'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engineC2EbPv[_ZN7webview17gtk_webkit_engineC5EbPv]+0x245): undefined reference to `webkit_settings_set_enable_developer_extras'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engineC2EbPv[_ZN7webview17gtk_webkit_engineC5EbPv]+0x255): undefined reference to `gtk_widget_show_all'
/usr/bin/ld: /home/coghi/Projetos/webview-tests/lib/webview/src/../ext/libwebview.a(webview.o): in function `webview::gtk_webkit_engine::run()':
webview.cc:(.text._ZN7webview17gtk_webkit_engine3runEv[_ZN7webview17gtk_webkit_engine3runEv]+0x11): undefined reference to `gtk_main'
/usr/bin/ld: /home/coghi/Projetos/webview-tests/lib/webview/src/../ext/libwebview.a(webview.o): in function `webview::gtk_webkit_engine::terminate()':
webview.cc:(.text._ZN7webview17gtk_webkit_engine9terminateEv[_ZN7webview17gtk_webkit_engine9terminateEv]+0x11): undefined reference to `gtk_main_quit'
/usr/bin/ld: /home/coghi/Projetos/webview-tests/lib/webview/src/../ext/libwebview.a(webview.o): in function `webview::gtk_webkit_engine::dispatch(std::function<void ()>)':
webview.cc:(.text._ZN7webview17gtk_webkit_engine8dispatchESt8functionIFvvEE[_ZN7webview17gtk_webkit_engine8dispatchESt8functionIFvvEE]+0x6c): undefined reference to `g_idle_add_full'
/usr/bin/ld: /home/coghi/Projetos/webview-tests/lib/webview/src/../ext/libwebview.a(webview.o): in function `webview::gtk_webkit_engine::set_title(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)':
webview.cc:(.text._ZN7webview17gtk_webkit_engine9set_titleENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN7webview17gtk_webkit_engine9set_titleENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x25): undefined reference to `gtk_window_get_type'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engine9set_titleENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN7webview17gtk_webkit_engine9set_titleENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x3b): undefined reference to `g_type_check_instance_cast'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engine9set_titleENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN7webview17gtk_webkit_engine9set_titleENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x46): undefined reference to `gtk_window_set_title'
/usr/bin/ld: /home/coghi/Projetos/webview-tests/lib/webview/src/../ext/libwebview.a(webview.o): in function `webview::gtk_webkit_engine::set_size(int, int, int)':
webview.cc:(.text._ZN7webview17gtk_webkit_engine8set_sizeEiii[_ZN7webview17gtk_webkit_engine8set_sizeEiii]+0x34): undefined reference to `gtk_window_get_type'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engine8set_sizeEiii[_ZN7webview17gtk_webkit_engine8set_sizeEiii]+0x4a): undefined reference to `g_type_check_instance_cast'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engine8set_sizeEiii[_ZN7webview17gtk_webkit_engine8set_sizeEiii]+0x54): undefined reference to `gtk_window_set_resizable'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engine8set_sizeEiii[_ZN7webview17gtk_webkit_engine8set_sizeEiii]+0x5f): undefined reference to `gtk_window_get_type'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engine8set_sizeEiii[_ZN7webview17gtk_webkit_engine8set_sizeEiii]+0x75): undefined reference to `g_type_check_instance_cast'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engine8set_sizeEiii[_ZN7webview17gtk_webkit_engine8set_sizeEiii]+0x88): undefined reference to `gtk_window_resize'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engine8set_sizeEiii[_ZN7webview17gtk_webkit_engine8set_sizeEiii]+0xab): undefined reference to `gtk_widget_set_size_request'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engine8set_sizeEiii[_ZN7webview17gtk_webkit_engine8set_sizeEiii]+0xdf): undefined reference to `gtk_window_get_type'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engine8set_sizeEiii[_ZN7webview17gtk_webkit_engine8set_sizeEiii]+0xf5): undefined reference to `g_type_check_instance_cast'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engine8set_sizeEiii[_ZN7webview17gtk_webkit_engine8set_sizeEiii]+0x10e): undefined reference to `gtk_window_set_geometry_hints'
/usr/bin/ld: /home/coghi/Projetos/webview-tests/lib/webview/src/../ext/libwebview.a(webview.o): in function `webview::gtk_webkit_engine::navigate(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)':
webview.cc:(.text._ZN7webview17gtk_webkit_engine8navigateENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN7webview17gtk_webkit_engine8navigateENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x25): undefined reference to `webkit_web_view_get_type'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engine8navigateENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN7webview17gtk_webkit_engine8navigateENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x3b): undefined reference to `g_type_check_instance_cast'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engine8navigateENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN7webview17gtk_webkit_engine8navigateENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x46): undefined reference to `webkit_web_view_load_uri'
/usr/bin/ld: /home/coghi/Projetos/webview-tests/lib/webview/src/../ext/libwebview.a(webview.o): in function `webview::gtk_webkit_engine::init(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)':
webview.cc:(.text._ZN7webview17gtk_webkit_engine4initENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN7webview17gtk_webkit_engine4initENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x15): undefined reference to `webkit_web_view_get_type'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engine4initENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN7webview17gtk_webkit_engine4initENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x2b): undefined reference to `g_type_check_instance_cast'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engine4initENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN7webview17gtk_webkit_engine4initENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x33): undefined reference to `webkit_web_view_get_user_content_manager'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engine4initENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN7webview17gtk_webkit_engine4initENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x60): undefined reference to `webkit_user_script_new'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engine4initENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN7webview17gtk_webkit_engine4initENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x72): undefined reference to `webkit_user_content_manager_add_script'
/usr/bin/ld: /home/coghi/Projetos/webview-tests/lib/webview/src/../ext/libwebview.a(webview.o): in function `webview::gtk_webkit_engine::eval(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)':
webview.cc:(.text._ZN7webview17gtk_webkit_engine4evalENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN7webview17gtk_webkit_engine4evalENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x25): undefined reference to `webkit_web_view_get_type'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engine4evalENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN7webview17gtk_webkit_engine4evalENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x3b): undefined reference to `g_type_check_instance_cast'
/usr/bin/ld: webview.cc:(.text._ZN7webview17gtk_webkit_engine4evalENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE[_ZN7webview17gtk_webkit_engine4evalENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE]+0x56): undefined reference to `webkit_web_view_run_javascript'
collect2: error: ld returned 1 exit status
Error: execution of command failed with code: 1: `cc "${@}" -o /home/coghi/.cache/crystal/crystal-run-url.tmp  -rdynamic -L/usr/bin/../lib/crystal `command -v pkg-config > /dev/null && pkg-config --cflags --libs gtk+-3.0 webkit2gtk-4.0` /home/coghi/Projetos/webview-tests/lib/webview/src/../ext/libwebview.a -lstdc++ -lpcre -lm -lgc -lpthread -levent -lrt -ldl`
```

</p></details>

--- a/src/lib.cr
+++ b/src/lib.cr
@@ -3,8 +3,8 @@ module Webview
     @[Link(framework: "WebKit")]
     @[Link(ldflags: "-L#{__DIR__}/../ext -lwebview.o -lc++")]
   {% elsif flag?(:linux) %}
-    @[Link(ldflags: "#{__DIR__}/../ext/libwebview.a -lstdc++")]
     @[Link(ldflags: "`command -v pkg-config > /dev/null && pkg-config --cflags --libs gtk+-3.0 webkit2gtk-4.0`")]
+    @[Link(ldflags: "#{__DIR__}/../ext/libwebview.a -lstdc++")]
   {% elsif flag?(:windows) %}
     # Windows requires special linker flags for GUI apps.
     @[Link(ldflags: "-L#{__DIR__}/../ext -lwebview.o -lc++")]

